mantidproject/mantid
# ISIS Powder GEM texture focusing - enable use of texture grouping file and save to .gem

Changes were added to allow GEM scientists to do texture focusing (focusing into 160 banks instead of 6):

- Add cropping values for the new 160-bank mode
- Add a routine to save to the `.gem` format (actually just the MAUD `.xye` format)
- Expose these with a new parameter, `texture_mode` on the GEM instrument object
- Implement a new algorithm to save a `grouping.new` grouping table
- Expose this with the parameter `save_angles`

**To test:**

As always, if not reviewing from ISIS, you'll need to change the second line and do some setup locally - ping me message for an explanation of how to set up ISIS_Powder directories, as it can be fiddly.

```
from isis_powder import Gem

gem = Gem(config_file=r"\\olympic\Babylon5\Public\JoeRamsay\GEM\Gem_config_example.yaml")

gem.focus(run_number=84420, # or any run number you like
          vanadium_normalisation=False,
          texture_mode=True,
          save_angles=True)
```

Click on a few children of `88420-ResultD` to make sure the data looks OK (as long as there's no NANs or anything, we should be fine). Make sure that `GEM84420_MAUD.gem` and `GEM84420_grouping.new` were written to `P:\JoeRamsay\GEM\Output\cycle_18_1\GEM`.

Fixes #22456 

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/commit/8e32a0c183a36fb39d20dea28935852db033e9ab)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
